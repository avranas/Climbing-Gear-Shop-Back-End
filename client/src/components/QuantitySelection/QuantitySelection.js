import { useEffect, useState } from "react";

//For use in the ProductPage

//Props are a handleSelection function
//A productOption
//Amount in Stock
const QuantitySelection = (props) => {

  const [ options, setOptions ] = useState([]);
  useEffect(() => {
    //Cap at amountInStock
    const newOptions = [];
    for(let i = 1; i <= props.amountInStock; i++) {
      newOptions.push(i);
    }
    setOptions(newOptions);
  }, [props.amountInStock])

  return (
    <div>
      <select
        onChange={props.handleSelection}
        name="quantity"
        id="quantity"
      >
        {
          options.map((i, key) => {
            return <option key={key} value={i}>{i}</option>
          })
        }
      </select>
    </div>
  );
};

export default QuantitySelection;











