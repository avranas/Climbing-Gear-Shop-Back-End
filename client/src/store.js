import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});