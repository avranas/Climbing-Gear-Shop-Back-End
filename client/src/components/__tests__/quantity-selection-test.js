import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import QuantitySelection from "../QuantitySelection/QuantitySelection";

afterEach(() => {
  cleanup();
});

const quantitySelection = (
  <Provider store={store}>
    <BrowserRouter>
      <QuantitySelection />
    </BrowserRouter>
  </Provider>
);

test("QuantitySelection matches snapshot", async () => {
  const tree = renderer.create(quantitySelection).toJSON();
  expect(tree).toMatchSnapshot();
});