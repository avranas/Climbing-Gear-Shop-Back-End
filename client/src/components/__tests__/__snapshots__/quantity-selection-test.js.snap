// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity matches snapshot 1`] = `
<div>
  <select
    className="form-select form-select-sm quantity-select"
    id="quantity"
    name="quantity"
    onChange={[Function]}
    value=""
  />
</div>
`;
